src/app/components/GoogleSheetsInput.svelte

```diff
@@ -0,0 +1,179 @@
+<script>
+  import worker from "../spawn-worker";
+
+  import InputField from "./InputField.svelte";
+  import Button from "./Button.svelte";
+
+  var spreadsheetId = "";
+  var error = null;
+
+  // Client ID and API key from the Developer Console
+  var CLIENT_ID = process.env.G_CLIENT_ID;
+  var API_KEY = process.env.G_API_KEY;
+
+  // Array of API discovery doc URLs for APIs used by the quickstart
+  var DISCOVERY_DOCS = [
+    "https://sheets.googleapis.com/$discovery/rest?version=v4",
+  ];
+
+  // Authorization scopes required by the API; multiple scopes can be
+  // included, separated by spaces.
+  var SCOPES = "https://www.googleapis.com/auth/spreadsheets.readonly";
+
+  // let authorizeButton = document.getElementById("authorize_button");
+  // let signoutButton = document.getElementById("signout_button");
+
+  /**
+   *  On load, called to load the auth2 library and API client library.
+   */
+  function handleClientLoad() {
+    gapi.load("client:auth2", initClient);
+  }
+
+  /**
+   *  Initializes the API client library and sets up sign-in state
+   *  listeners.
+   */
+  function initClient() {
+    error = null;
+    gapi.client
+      .init({
+        apiKey: API_KEY,
+        clientId: CLIENT_ID,
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES,
+      })
+      .then(
+        function () {
```

Comment:
```suggestion
        ()=> 
```
Since you are resolving a promise here, using a `then catch` would make the code more descriptive